Comgr fix (#3801)
ROCm-Developer-Tools/HIP: docs/how-to/hip_rtc.rst

  * This library can be used for compilation on systems without AMD GPU drivers
    installed (offline compilation). However, running the compiled code still
    requires both the HIP runtime library and GPU drivers on the target system.
  * Developers can bundle this library with their application.

>   * hipRTC leverages AMD's Code Object Manager API (``Comgr``) internally, which

Reviewer: better use all capitalized HIPRTC to be consistent with https://github.com/ROCm/hip/blob/amd-staging/docs/how-to/hip_rtc.rst

>   * hipRTC leverages AMD's Code Object Manager API (``Comgr``) internally, which

Author: ```suggestion
  * HIPRTC leverages AMD's Code Object Manager API (``Comgr``) internally, which
```

    is designed to simplify linking, compiling, and inspecting code objects. For

>     more information, see the `llvm-project/amd/comgr/README <https://github.com/ROCm/llvm-project/amd-staging/amd/comgr/README.md>`_.

Reviewer 2: Is the `_` a typo ?

Reviewer 2: @randyh62 what abou this one ?

Author: No, it is required for the reference in RST format. If you look at the rendered HTML it is correct.

Reviewer 2: Gotcha

> 

Reviewer 3: ```suggestion
* Comgr may cache hipRTC compilations. To force full recompilation for each hipRTC API invocation, set AMD_COMGR_CACHE=0.
```

> * Comgr may cache hipRTC compilations. To force full recompilation for each hipRTC API invocation, set AMD_COMGR_CACHE=0.

Author: ```suggestion
  * Comgr may cache HIPRTC compilations. To force full recompilation for each HIPRTC API invocation, set AMD_COMGR_CACHE=0.
```

    - When viewing the *README* in the Comgr GitHub repository you should look at a

>       specific release of interest rather than the default branch.

Reviewer 2: ```suggestion
      [specific release](https://github.com/ROCm/llvm-project/tree/release/rocm-rel-6.4/amd/comgr) of interest rather than the default branch.
```

Reviewer 2: @randyh62 could we include this as well ?

Reviewer 2: I guess this is Markdown formatting, but equivalent for RST formatting

  * This library can be used for compilation on systems without AMD GPU drivers
    installed (offline compilation). However, running the compiled code still
    requires both the HIP runtime library and GPU drivers on the target system.
  * Developers can bundle this library with their application.
  * HIPRTC leverages AMD's Code Object Manager API (``Comgr``) internally, which
    is designed to simplify linking, compiling, and inspecting code objects. For

>     more information, see the `llvm-project/amd/comgr/README <https://github.com/ROCm/llvm-project/blob/amd-staging/README.md>`_.

Author: ```suggestion
    more information, see the `llvm-project/amd/comgr/README <https://github.com/ROCm/llvm-project/blob/amd-staging/amd/comgr/README.md>`_.
```

>     more information, see the `llvm-project/amd/comgr/README <https://github.com/ROCm/llvm-project/amd-staging/amd/comgr/README.md>`_.

Author: ```suggestion
    more information, see the `llvm-project/amd/comgr/README <https://github.com/ROCm/llvm-project/blob/amd-staging/README.md>`_.
```

  * Comgr may cache HIPRTC compilations. To force full recompilation for each HIPRTC API invocation, set AMD_COMGR_CACHE=0.

>     - When viewing the *README* in the Comgr GitHub repository you should look at a
>       specific release of interest rather than the default branch.

Author: ```suggestion
    - When viewing the *README* in the Comgr GitHub repository you should look at a
      specific branch of interest, such as docs/6.3.0 or docs/6.4.1, rather than the default branch.
```

>       specific branch of interest, such as docs/6.3.0 or docs/6.4.1, rather than the default branch.

Author: ```suggestion
      specific branch of interest, such as ``docs/6.3.0`` or ``docs/6.4.1``, rather than the default branch.
```

  * This library depends on Code Object Manager (comgr). You can try to

>     statically link comgr into HIPRTC to avoid ambiguity.

Reviewer 2: If we are adding the new point that describes how hipRTC uses COMGR, we can probably remove this point.

disabled, the location for storing the cache contents, and the cache eviction
policy:

* ``AMD_COMGR_CACHE`` By default this variable has a value of ``1`` and the compilation
cache feature is enabled. To disable the feature set the environment variable to

> a value of ``0``.

Reviewer 2: This change will be available with ROCm 7.0 only. Do the docs go live with the release or should we hold out on this change?

installed AMD compiler, it must run with HIPRTC and comgr dynamic libraries that
are compatible with the version of the bitcode/IR.

`Comgr <https://github.com/ROCm/llvm-project/tree/amd-staging/amd/comgr/README.md>`_ is a
shared library that incorporates the LLVM/Clang compiler that HIPRTC relies on.
To identify the bitcode/IR version that comgr is compatible with, one can
execute "clang -v" using the clang binary from the same ROCm or HIP SDK package.
For instance, if compiling bitcode/IR version 14, the HIPRTC and comgr libraries
released by AMD around mid 2022 would be the best choice, assuming the
LLVM/Clang version included in the package is also version 14.

.. note:: 

>   When viewing the *README* in Comgr GitHub respository, you should look at a specific
>   release of interest rather than the default branch. 

Author: ```suggestion
  When viewing the *README* in the Comgr GitHub repository you should look at a
  specific release of interest, such as docs/6.3.0 or docs/6.4.1, rather than the default branch.
```

>   specific release of interest, such as docs/6.3.0 or docs/6.4.1, rather than the default branch.

Author: ```suggestion
  specific branch of interest, such as ``docs/6.3.0`` or ``docs/6.4.1``, rather than the default branch.
```

installed AMD compiler, it must run with HIPRTC and comgr dynamic libraries that
are compatible with the version of the bitcode/IR.

`Comgr <https://github.com/ROCm/llvm-project/tree/amd-staging/amd/comgr/README.md>`_ is a
shared library that incorporates the LLVM/Clang compiler that HIPRTC relies on.
To identify the bitcode/IR version that comgr is compatible with, one can
execute "clang -v" using the clang binary from the same ROCm or HIP SDK package.
For instance, if compiling bitcode/IR version 14, the HIPRTC and comgr libraries
released by AMD around mid 2022 would be the best choice, assuming the
LLVM/Clang version included in the package is also version 14.

.. note:: 
  When viewing the Comgr GitHub respository, you should look at a specific
  release of interest rather than the default branch. 

> 

Reviewer 2: We could add this link as an example: https://github.com/ROCm/llvm-project/tree/release/rocm-rel-6.4
